format-common/src/main/java/io/cdap/plugin/format/charset/fixedlength/FixedLengthCharsetTransformingDecompressorStream.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.charset.fixedlength;
+
+import org.apache.hadoop.io.compress.DecompressorStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * DecompressorStream implementation for the FixedLengthCharsetTransformingDecompressor.
+ * <p>
+ * This can be used to convert input streams containing bytes for fixed length charsets into UTF-8 bytes.
+ */
+public class FixedLengthCharsetTransformingDecompressorStream extends DecompressorStream {
+  private static final Logger LOG = LoggerFactory.getLogger(FixedLengthCharsetTransformingDecompressorStream.class);
+
+  //Starting and ending position in the file.
+  long start;
+  long end;
+
+  protected FixedLengthCharsetTransformingDecompressorStream(InputStream in,
+                                                             FixedLengthCharset fixedLengthCharset,
+                                                             long start,
+                                                             long end)
+    throws IOException {
+    super(in, new FixedLengthCharsetTransformingDecompressor(fixedLengthCharset));
+    long skippedBytes = in.skip(start);
```

Comment:
unused?

Reply:
Indeed, we're not really using this value. Maybe I should throw an exception if it's not possible to follow this instruction.

Reply:
I believe getting less than asked means we've got EOF earlier.